### DISCUSSION TITLE
How does viz.py and other files retrieve data from the database?

### DISCUSSION THREAD

**Discussion by JarethRajan** [apache/superset#12914] [Q&A / Help] [answered]
My company and I are currently using a modified Apache Superset version 0.38.0, but still uses all the fundamentals. 

**I am unsure as to how and where the system is retrieving the values from the databases, as there seems to be no clear link.**

Setting "useLegacyApi" in metadata for a chart determines if it uses viz.py, but I am unsure on how to trace back to find the relevant information and files that connect it to the database.
Besides viz.py,  "/api/v1/query/" is what is used if "useLegacyApi" is set to false. I am also unsure as to how this connects to the database.

Some help regarding this and data passing would be grateful.

**Comment by ktmud** [comment, marked as answer]
Hi, it's a little convoluted, but I'd recommend checking out these lines:

https://github.com/apache/superset/blob/32f2c45f93693156f1cf88c2c5223684c31d3f20/superset/viz.py#L501

https://github.com/apache/superset/blob/32f2c45f93693156f1cf88c2c5223684c31d3f20/superset/viz.py#L259

https://github.com/apache/superset/blob/32f2c45f93693156f1cf88c2c5223684c31d3f20/superset/viz.py#L275

https://github.com/apache/superset/blob/dcef5e1381e6a089264784a5293dfc380524024e/superset/connectors/sqla/models.py#L1297

You should also spend some time to configure you code editor/IDE to have proper code jumps: https://code.visualstudio.com/docs/editor/editingevolved

and Python debugger: https://code.visualstudio.com/docs/python/tutorial-flask

It will save you a lot of time down the road!

**Reply by JarethRajan** [reply]
Thanks! I'm still learning how the files interact with one another, so stuff like this eludes me. 
I'll try to update here if I find what I'm looking for, or if I have anymore questions.
